src/core/core.cc

```diff
@@ -109,41 +109,28 @@ namespace SSC {
-      "                                                              \n"
-      "  try {                                                       \n"
-      "    params = JSON.parse(params);                              \n"
-      "  } catch (err) {                                             \n"
-      "    console.error(err.stack || err, params);                  \n"
-      "  };                                                          \n"
-      "                                                              \n"
-      "  const headers = `" + trim(post.headers) + "`                \n"
-      "    .trim()                                                   \n"
-      "    .split(/[\\r\\n]+/)                                       \n"
-      "    .filter(Boolean);                                         \n"
-      "                                                              \n"
-      "  const detail = {                                            \n"
-      "    data: xhr.response,                                       \n"
-      "    sid: '" + sid + "',                                       \n"
-      "    headers: Object.fromEntries(                              \n"
-      "      headers.map(l => l.split(/\\s*:\\s*/))                  \n"
-      "    ),                                                        \n"
-      "    params: params                                            \n"
-      "  };                                                          \n"
-      "                                                              \n"
-      "  queueMicrotask(() => {                                      \n"
-      "    const event = new window.CustomEvent('data', { detail }); \n"
-      "    window.dispatchEvent(event);                              \n"
-      "  });                                                         \n"
-      "};                                                            \n"
-      "                                                              \n"
-      "xhr.open('GET', `ipc://post?id=" + sid + "`);                 \n"
-      "xhr.setRequestHeader('Access-Control-Allow-Origin', '*');     \n"
-      "xhr.send();                                                   \n"
+      "const globals = await import('socket:internal/globals');              \n"
+      "const id = `" + sid + "`;                                             \n"
+      "const seq = `" + seq + "`;                                            \n"
+      "const headers = `" + trim(post.headers) + "`                          \n"
+      "  .trim()                                                             \n"
+      "  .split(/[\\r\\n]+/)                                                 \n"
+      "  .filter(Boolean);                                                   \n"
+      "                                                                      \n"
+      "let params = `" + params + "`;                                        \n"
```

Comment:
does this need to escape backticks?

Reply:
it shouldn't! I just copied this from what was there before